frontend/src/components/WofButton.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <button :class="`button button--${type}`" :style="`font-size: ${this.size}rem;`" @click="onClick()">
+    <wof-icon v-if="this.icon" :icon="this.icon" :size="this.size" class="button--icon" />
+    <slot/>
+  </button>
+</template>
+
+<script>
+export default {
+  name: "WofButton",
+
+  props: {
+    type: {
+      type: String,
+      required: true,
+      default: "default",
+      validator(value) {
+        return ["default", "positive", "warning", "error", "disabled"].includes(value)
+      }
+    },
+    size: {
+      type: Number,
+      required: false,
+      default: 1
+    },
+    icon: {
+      type: String,
+      required: false
+    }
+  },
+  methods: {
+    onClick: function(event) {
+      this.$emit("click", event)
+    }
+  }
+}
+</script>
+
+<style lang="less">
+@import './common.less';
+
+.button {
+  border: none;
+  height: min-content;
+  text-align: center;
+  align-items: center;
+  display: flex;
+  border-radius: 1rem;
+  padding: 0.5rem 1rem;
+  color: @primary-bright-text-color;
+  font-weight: 600;
+  &:active {
+    color: @primary-dark-text-color;
+  }
+  &--icon {
```

Comment:
The icon is a button element, not a modifier.
So class for icon should be 'button__icon'